src/specBuilder/signal/signalSpecBuilder.ts

```diff
@@ -94,3 +97,95 @@ export const getLegendLabelsSeriesSignal = (value: unknown = null): Signal => {
 export const getGenericSignal = (name: string, value: unknown = null): Signal => {
 	return { name, value };
 };
+
+//TODO: Add documentation
+export const getAnimationSignals = (name: string): Signal[] => {
+	return [
+		getRSCAnimation(),
+		getRSCColorAnimationDirection(name),
+		getRSCColorAnimation(),
+		getPrevHoverSignal(name),
+		getSelectedID(name)
+	]
+}
+
+export const getRSCLegendColorAnimationDirection = (name: string): ({ update: string; events: string })[] => {
+	return [
+		{ events: `@${name}_legendEntry:mouseover`, update: '1' },
+		{ events: `@${name}_legendEntry:mouseout`, update: '-1' }
+	];
+};
+
+//TODO: add documentation
+const getRSCAnimation = (): Signal => {
+	return {
+		name: 'rscAnimation',
+		value: 0,
+		on: [{
+			events: 'timer{16}',
+			update: 'scale(\'rscAnimationCurve\', scale(\'rscAnimationCurveInverse\', rscAnimation) + 0.03333333333333334)'
+		}]
+	};
+};
+
+//TODO: add documentation
+const getRSCColorAnimationDirection = (name: string): Signal => {
+	if (name === 'line0') {
```

Comment:
Using `name` as a variable here can be confusing, as it visually conflicts with the object key called `name`